polygon-digital-carbon/src/TransferHandler.ts

```diff
@@ -101,13 +96,16 @@ export function handleToucanPuroRetirementRequested(event: RetirementRequested):
 }
 
 export function handleToucanPuroRetirementFinalized(event: RetirementFinalized): void {
-  let request = loadOrCreateToucanBridgeRequest(event.params.requestId)
-  request.status = 'FINALIZED'
+  let requestIdByteArray = Bytes.fromBigInt(event.params.requestId)
+  let requestId = Bytes.fromByteArray(requestIdByteArray)
+
+  let request = loadOrCreateAsyncRetireRequest(requestId)
+  request.status = AsyncRetireRequestStatus.FINALIZED
   request.save()
 
   // if (request.retire !== null) {
```

Comment:
Since the `Retire` entity is now non-nullable, should this be un-commented and the status updated? On the current test deployment the Puro retirement requested via tx hash 0xdbe415ed2b4e1e86e7a545c753e4f9dc2d325fa1107ec282cf2017f262b8fcb3 has a Finalized status under `asyncRetireRequest`, and a Requested status under `Retire`

Reply:
fixed